src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -84,69 +91,20 @@ pub fn modify(sess: &ParseSess,
     return krate
 }
 
-impl<'a> CollectCustomDerives<'a> {
+impl<'a> CollectProcMacros<'a> {
     fn check_not_pub_in_root(&self, vis: &ast::Visibility, sp: Span) {
         if self.is_proc_macro_crate &&
            self.in_root &&
            *vis == ast::Visibility::Public {
             self.handler.span_err(sp,
                                   "`proc-macro` crate types cannot \
                                    export any items other than functions \
-                                   tagged with `#[proc_macro_derive]` \
+                                   tagged with `#[proc_macro_derive]` or `#[proc_macro_attribute]` \
```

Comment:
I think we should avoid changing the diagnostics for users of stable macros 1.1 custom derives.